static/js/browser2/browser.ts

```diff
@@ -19,22 +19,52 @@ import ReactDOM from "react-dom";
 import _ from "lodash";
 
 import { BrowserPage, nodeTypeEnum } from "./browser_page";
+import axios from "axios";
+import { removeLoadingMessage } from "./shared";
 
 window.onload = () => {
   const dcid = document.getElementById("node").dataset.dcid;
   const nodeName = document.getElementById("node").dataset.nn;
   const urlParams = new URLSearchParams(window.location.search);
   const statVarId = urlParams.get("statVar") || "";
-  const nodeType = _.isEmpty(statVarId)
-    ? nodeTypeEnum.GENERAL
-    : nodeTypeEnum.PLACE_STAT_VAR;
-  ReactDOM.render(
-    React.createElement(BrowserPage, {
-      dcid,
-      nodeType,
-      nodeName,
-      statVarId,
-    }),
-    document.getElementById("node")
-  );
+  axios
+    .get(`/api/browser/propvals/typeOf/${dcid}`)
+    .then((resp) => {
+      const values = resp.data.values;
+      const types = values.out ? values.out : [];
+      const listOfTypes = types.map((type) => type.dcid);
+      ReactDOM.render(
+        React.createElement(BrowserPage, {
+          dcid,
+          nodeType: getNodeType(listOfTypes, statVarId),
+          nodeName,
+          statVarId,
+        }),
+        document.getElementById("node")
+      );
+    })
+    .catch(() => removeLoadingMessage());
 };
+
+function getNodeType(listOfTypes, statVarId): string {
```

Comment:
this util belongs together with the definition of nodeTypeEnum. and perhaps both belong to shared.ts (or a new types.ts)

i also think it would be much clearer if this strictly returns nodetypeEnum (and that should not map to string and perhaps be a strict numeric enum). this now introduces a few new node types, and it's not really clear what the universe of return values could be. 

additionally, i think this function would be much simpler if there was a prioritized mapping for nodeType (from kg) to pageDisplayType (for what type of browser page to display). then this is a simple matter of walking down the map and returning the first match without any special cases.

please also add a one-liner comment describing the function

Reply:
done. Changed how to figure out the type of page to render for a nodeType